src/WebJobs.Script/Rpc/LanguageWorkerProcess.cs

```diff
@@ -184,8 +190,9 @@ internal void HandleWorkerProcessExitError(LanguageWorkerProcessExitException la
             // The subscriber of WorkerErrorEvent is expected to Dispose() the errored channel
             if (langExc != null && langExc.ExitCode != -1)
             {
-                _workerProcessLogger.LogDebug(langExc, $"Language Worker Process exited.", _process.StartInfo.FileName);
-                _eventManager.Publish(new WorkerErrorEvent(_runtime, _workerId, langExc));
+                _workerProcessLogger.LogDebug(langExc, $"Language Worker Process exited (exit code: {langExc.ExitCode}).", _process.StartInfo.FileName);
+                var isIntentionalExit = langExc.ExitCode == NonErrorExitCode;
+                _eventManager.Publish(new WorkerErrorEvent(_runtime, _workerId, isIntentionalExit ? null : langExc));
```

Comment:
If this is a graceful exit of the process, can this logic move to OnProcessExited and thereby do _process.WaitForExit(); and _process.Close() before publishing a new type of event (WorkerRestartEvent)?

Reply:
@yojagad Yes, introducing a separate event may also be a good idea (unless we just do what Pragna suggested). I wanted to piggyback to `WorkerErrorEvent` just to reduce the amount of code changes, but this may be cleaner. Let's quickly sync offline.

Reply:
I agree with this approach. Cleaner. Publishing WorkerRestartEvent would simply not count towards language worker errors.

Reply:
Discussed offline, decided to proceed with a new event.